packages/devtools/lib/src/ui/service_extension_elements.dart

```diff
@@ -177,15 +177,49 @@ class ServiceExtensionSelector {
     _selectedValue = selector.value;
   }
 
-  void _updateState() {
+  void addOptions() {
+    extensionDescription.displayValues.forEach(selector.option);
+  }
+
+  void updateState() {
+    // Select option whose state is already enabled.
+    serviceManager.serviceExtensionManager
+        .getServiceExtensionState(extensionDescription.extension, (state) {
+      updateSelection(state);
+    });
+  }
+
+  void updateSelection(ServiceExtensionState state) {
+    if (state.value != null) {
+      final selectedIndex = extensionDescription.values.indexOf(state.value);
+      selector.selectedIndex = selectedIndex;
+      _selectedValue = extensionDescription.displayValues[selectedIndex];
+    }
+  }
+}
+
+class TogglePlatformSelector extends ServiceExtensionSelector {
+  TogglePlatformSelector() : super(togglePlatformMode);
+
+  static const fuchsia = 'Fuchsia';
+
+  @override
+  void addOptions() {
+    extensionDescription.displayValues
+        .where((displayValue) => !displayValue.contains(fuchsia))
+        .forEach(selector.option);
+  }
+
+  @override
+  void updateState() {
     // Select option whose state is already enabled.
     serviceManager.serviceExtensionManager
         .getServiceExtensionState(extensionDescription.extension, (state) {
-      if (state.value != null) {
-        final selectedIndex = extensionDescription.values.indexOf(state.value);
-        selector.selectedIndex = selectedIndex;
-        _selectedValue = extensionDescription.displayValues[selectedIndex];
+      if (state.value == fuchsia.toLowerCase()) {
+        selector.option(extensionDescription.displayValues
```

Comment:
what if the fuchsia option isn't present in the list?

Reply:
i would expect firstWhere throws an exception in that case which isn't what you want .

Reply:
We control the state of the extensionDescription, so it will always be in the list unless we change it. Optionally, we can also grab `extensionDescription.displayValues.last`, but I did firstWhere to match on name rather than index.